core/locks_hasher.go

```diff
@@ -0,0 +1,61 @@
+package core
+
+import (
+	"context"
+	"fmt"
+	"hash"
+	"hash/fnv"
+)
+
+const (
+	DefaultLockConcurrency = 1024
+	DefaultLockIdentifier  = "default-lock"
+)
+
+type (
+	LockHasher struct {
+		ctx         context.Context
+		concurrency uint32
+		locks       [DefaultLockConcurrency]*LockH
+	}
+)
+
+func NewLockHasher() (lockHsh *LockHasher) {
+	lockHsh = &LockHasher{
+		ctx:         context.Background(),
+		concurrency: uint32(DefaultLockConcurrency),
+	}
+	for i := 0; i < DefaultLockConcurrency; i++ {
+		lockHsh.locks[i] = NewLockH()
+	}
+	return
+}
+
+func (lockHsh *LockHasher) GetHashKey(strKey string) (hashSlot uint32, err error) {
+	var (
+		hashFn hash.Hash32
+	)
+	if strKey == "" {
+		strKey = DefaultLockIdentifier
+	}
+	hashFn = fnv.New32a()
+	if _, err = hashFn.Write([]byte(strKey)); err != nil {
+		return
+	}
+	hashSlot = hashFn.Sum32() % lockHsh.concurrency
+	return
+}
+
+func (lockHsh *LockHasher) GetStore(strKey string) (lockH *LockH, err error) {
```

Comment:
GetHashKey => GetHash, GetStore => GetLock wdyt?

Reply:
Renamed GetHashKey to GetHash.
Renamed GetStore to GetLockStore.